### PR TITLE
Sidebar polishing

### PR DESCRIPTION
**Pull request by juliusknorr**
Fixes:
- The currently selected file is not highlighted in the file list anymore https://github.com/nextcloud/server/issues/17737
- The tags should not me shown if none are set https://github.com/nextcloud/server/issues/17737
- Renaming a file opens the sidebar https://github.com/nextcloud/server/issues/18613

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Tested and fixes the mentioned points!
:guitar:

**Comment by skjnldsv** [comment]
> * The tags should not me shown if none are set #17737

How do we toggle the view then?

**Comment by rullzer** [comment]
> > * The tags should not me shown if none are set #17737
> 
> How do we toggle the view then?

You click the tag button ;)

**Comment by rullzer** [comment]
tags

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -650,8 +650,18 @@
...
+			}
+			this.$fileList.children().removeClass('highlighted');
+			tr.addClass('highlighted');
+			this._currentFileModel = model;
+
 			// open sidebar and set file
-			OCA.Files.Sidebar.open(path.replace('//', '/'))
+			if (typeof show === 'undefined' || !!show) {
+				OCA.Files.Sidebar.open(path.replace('//', '/'))
+			}
```
Comment (gary-kim) [inline apps/files/js/filelist.js]: Even if `show === false`, the sidebar should be opened to the new file if the Sidebar was already open. How about this?
```javascript
(typeof show === 'undefined' || !!show || OCA.Files.Sidebar.file !== '')
```

**Review by gary-kim — CHANGES_REQUESTED** [review]

File: apps/files/js/filelist.js (hunk 1 above)
Comment (juliushaertl) [reply apps/files/js/filelist.js]: Wouldn't it then be just called with `_updateDetailsView(fileName)` and therefore call the open method?

**Comment by skjnldsv** [comment]
Ideally it would have be nice to have a button in the sidebar Actions slot to show the tags. At least that is how Jan and I decided last time we talked about this :see_no_evil:

File: apps/files/js/filelist.js (hunk 1 above)
Comment (gary-kim) [reply apps/files/js/filelist.js]: It's used on renaming files, so if the sidebar was already open when a file is being renamed, switch to that file in the sidebar. If it was closed, don't open it.

From #18613

https://github.com/nextcloud/server/blob/27dc4ef268722f525b3ce1f1fe72f0b17531ad6a/apps/files/js/filelist.js#L2662-L2665

**Comment by rullzer** [comment]
master is NC19 now.
Please backport if it needs to go into 18

**Comment by jancborchardt** [comment]
> Ideally it would have be nice to have a button in the sidebar Actions slot to show the tags. At least that is how Jan and I decided last time we talked about this see_no_evil

Yes please :) :label:

**Review by gary-kim — APPROVED** [review]

**Comment by gary-kim** [comment]
@jancborchardt @skjnldsv Want to take another look at the tags?

**Comment by skjnldsv** [comment]
Should be good to go :)

**Review by skjnldsv — APPROVED** [review]

**Review by jancborchardt — CHANGES_REQUESTED** [review]
- [x] When clicking "Manage tags", the "Collaborative tags" field should be focused and the popover of the 3-dot-menu should be closed
- [x] When the tags field is already shown and you click "Manage tags", the popover should be closed after hiding the tags field
- [x] I’d say simplify it from "Manage tags" to "Tags" since that is more concise and will work better with internationalization as well

**Comment by skjnldsv** [comment]
Backport in https://github.com/nextcloud/server/pull/18911

**Pull request merged**